apps/server/src/modules/h5p-editor/contentStorage/contentStorage.ts

```diff
@@ -1,173 +1,156 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
-import { Injectable } from '@nestjs/common';
-import { Readable, Stream } from 'stream';
-import * as fs from 'node:fs';
-import * as fsPromises from 'node:fs/promises';
-import { getAllFiles } from 'get-all-files';
 import {
 	ContentId,
 	IContentMetadata,
 	IContentStorage,
-	LibraryName,
 	IFileStats,
 	ILibraryName,
 	IUser,
+	LibraryName,
 	Permission,
 	streamToString,
 } from '@lumieducation/h5p-server';
-import path from 'path';
-import rimraf from 'rimraf';
+import { Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { FileDto } from '@src/modules/files-storage/dto';
+import { Readable } from 'stream';
+import { S3ClientAdapter } from '../../files-storage/client/s3-client.adapter';
+import { H5PContent } from './h5p-content.entity';
+import { H5PContentRepo } from './h5p-content.repo';
 
 @Injectable()
 export class ContentStorage implements IContentStorage {
-	constructor(
-		protected contentPath: string,
-		protected options?: {
-			invalidCharactersRegexp?: RegExp;
-			maxPathLength?: number;
-		}
-	) {}
+	constructor(private readonly repo: H5PContentRepo, private readonly storageClient: S3ClientAdapter) {}
 
 	public async addContent(
 		metadata: IContentMetadata,
-		content: unknown,
+		content: string | undefined,
 		user: IUser,
 		contentId?: ContentId | undefined
 	): Promise<ContentId> {
-		if (contentId === null || contentId === undefined) {
-			contentId = await this.createContentId();
+		if (!metadata.defaultLanguage) {
+			metadata.defaultLanguage = metadata.language;
 		}
+
 		try {
-			fs.existsSync(path.join(this.getContentPath(), contentId.toString()));
-			await this.existsOrCreateDir(contentId);
-			await fsPromises.writeFile(
-				path.join(this.getContentPath(), contentId.toString(), 'h5p.json'),
-				JSON.stringify(metadata)
-			);
-			await fsPromises.writeFile(
-				path.join(this.getContentPath(), contentId.toString(), 'content.json'),
-				JSON.stringify(content)
-			);
-		} catch (error) {
-			if (fs.existsSync(path.join(this.getContentPath(), contentId.toString()))) {
-				rimraf.sync(path.join(this.getContentPath(), contentId.toString()));
+			let h5pContent: H5PContent;
+
+			if (contentId) {
+				h5pContent = await this.repo.findById(contentId);
+				h5pContent.metadata = metadata;
+				h5pContent.content = content;
+				await this.repo.save(h5pContent);
+			} else {
+				h5pContent = new H5PContent({ metadata, content });
+				await this.repo.createContent(h5pContent);
```

Comment:
The MikroORM docs say "we instantiate new entity via new Author() or em.create(Author, { ... })". (https://mikro-orm.io/docs/defining-entities#default-values) So you don't need to do both in a row.

You can also get rid of that repo method then, as it is not used anywhere else.